### PR TITLE
Purge simulation manager

### LINKED ISSUE #64: Replace mentions of BACI with 4C
Replace all occurrences of BACI with 4C. This requires that we wait until everything in the BACI repository is renamed to 4C, e.g., executables, ... .

### PR DESCRIPTION
**Pull request by isteinbrecher**
Closes #64

### CODE REVIEW

**Comment by isteinbrecher** [comment]
@knarfnitram this is a sort of cleaned up variant of a local branch of mine. This might not be the final thing that will be merged, but this should somewhat show how the "run_4C" functionality could look in the future.

**Comment by isteinbrecher** [comment]
@knarfnitram I think this is now ready to be merged / reviewed. This merge request removes the simulation manager and adds a function `run_four_c` which allows to run 4C from a function. The relevant parameters can also be set via the env variables `MESHPY_FOUR_C_EXE`, `MESHPY_MPI_COMMAND` and `MESHPY_MPI_NUM_PROC`, e.g., if this is used inside a QUEENS run.

File: meshpy/four_c/run_four_c.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    restart_step=None,
+    restart_from=None,
+    log_to_console=False,
+):
+    """Run a 4C simulation and return the exit code of the run
+
+    This function looks into the environment variables for some parameters:
+        "MESHPY_FOUR_C_EXE"
+        "MESHPY_MPI_COMMAND"
+        "MESHPY_MPI_NUM_PROC"
```
Comment (knarfnitram) [inline meshpy/four_c/run_four_c.py:56]: Do we have to update the environment variables in the `README.md` ?

File: meshpy/four_c/run_four_c.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        Return code of 4C run
+    """
+
+    def get_env_variable(name, *, fail_if_not_set=False, default=None):
+        """Return the value of an environment variable"""
+        if name in os.environ.keys():
+            return os.environ[name]
+        elif fail_if_not_set:
+            raise ValueError(f"Environment variable {name} is not set")
+        return default
```
Comment (knarfnitram) [inline meshpy/four_c/run_four_c.py:94]: Should we consider moving this function to a more central place?

**Review by knarfnitram — DISMISSED** [review]

File: meshpy/four_c/run_four_c.py (hunk 2 above)
Comment (isteinbrecher) [reply meshpy/four_c/run_four_c.py:94]: Good idea, I will do that.

**Pull request merged**